### DISCUSSION TITLE
Better approach

### DISCUSSION THREAD

**Discussion by khanra17** [ReVanced/revanced-discussions#387] [Question] [answered]
Compiling every patch in a single jar sounds a bad idea.
It would be good if pathes are independent & hosted in a repository.
Every patch has a yaml file that has metadata like dependencies of other patches.

Also compiling paches limit it's reach.
I think a neat, simple & modular approach would be _

- ByteCodePatches 
  - Could be a text file containing regex (Maybe embedded in the yaml)
- ResourcePatch
  - additional resources are zipped and instructions are inside yaml

**Comment by oSumAtrIX** [comment, marked as answer]
Bytecode patches are more advanced and make use of Dalvik assembling and disassembling. Patching raw bytes like Lucky Patcher is not viable.

**Reply by oSumAtrIX** [reply]
Additionally, nothing stops you to create a patch per JAR file, but it's not practical.

**Reply by khanra17** [reply]
No don't mean raw binary but smali patches,
Also  I think per patch is more supirior than speghetting everything in one.
It's in the early stage thats why its manageable now but sooner or leter a repository hosting patches is the solution

**Reply by oSumAtrIX** [reply]
SMALI is also not all we do, we do much more than that. I recommend looking at existing patches.  Additionally, it's not spaghetti in one JAR as it's structured by packages. Also, a repository IS already hosting the patches.
